### DISCUSSION TITLE
Generate dataset of female patients with COVID produces only deceased patients

### DISCUSSION THREAD

**Discussion by davide-aloi** [synthetichealth/synthea#1418] [Q&A] [answered]
Dear Synthea users,

I am trying to generate a dataset including only female patients that were diagnosed with COVID19. To achieve such a dataset I am running synthea with the following parameters:

```
./run_synthea -s 12345 \
    -p 200 \
    -g F \
    -m *covid19 \
    -k keep.json \
    --exporter.baseDirectory='/data' \
    --exporter.csv.export.conditions=true \ 
```

The file keep.json was generated via https://synthetichealth.github.io/spt/#/customizer and has the following content:

```
{
  "name": "Generated Keep Module",
  "states": {
    "Initial": {
      "type": "Initial",
      "name": "Initial",
      "conditional_transition": [
        {
          "transition": "Keep",
          "condition": {
            "condition_type": "And",
            "conditions": [
              {
                "condition_type": "Active Condition",
                "codes": [
                  {
                    "system": "SNOMED-CT",
                    "code": "840539006",
                    "display": "COVID-19"
                  }
                ]
              }
            ]
          }
        },
        {
          "transition": "Terminal"
        }
      ]
    },
    "Terminal": {
      "type": "Terminal",
      "name": "Terminal"
    },
    "Keep": {
      "type": "Terminal",
      "name": "Keep"
    }
  },
  "gmf_version": 2
}
```

When I don't use the above keep.json, I get a sample of 200 participants, some with a diagnosed of covid and some without, and not all of them are deceased. However, when I add the keep.json, all the patients that are being generated are deceased. Is there anything that has to be specified in the keep.json to get both of alive/deceased patients? 

Many thanks for the help!

**Comment by dehall** [comment, marked as answer]
I believe the issue is that this keep module will only keep patients with an *active* diagnosis of COVID-19 (as of the end of the simulation == the current date), and the infection rates are now fairly low, plus the duration of infection is also fairly short. 
One "quick fix" to get more patients would be to set an end date of the simulation in 2020 or 2021 for example, with the `-e` flag: `-e 20200601` .
Another option, if you're ok with patients that have *ever* had covid19, not necessarily currently active, you can change the keep module to use the `covid19` attribute instead and check for "not nil", see below.

If neither of these works for you, the last option is to modify the covid19 module infection rates, but that's a more involved process. I can provide some pointers if necessary

```
{
  "name": "Generated Keep Module",
  "states": {
    "Initial": {
      "type": "Initial",
      "conditional_transition": [
        {
          "transition": "Keep",
          "condition": {
            "condition_type": "Attribute",
            "attribute": "covid19",
            "operator": "is not nil"
          }
        },
        {
          "transition": "Terminal"
        }
      ]
    },
    "Terminal": {
      "type": "Terminal"
    },
    "Keep": {
      "type": "Terminal"
    }
  },
  "gmf_version": 2
}
```

**Reply by davide-aloi** [reply]
Many thanks, the above keep.json worked perfectly.
